### ISSUE TITLE
Exported variable with custom getter makes Debugger unresponsive when there's infinite recursion

### ISSUE THREAD

**Issue by L4Vo5** [godotengine/godot#79230]
### Godot version

v4.1.stable.official [970459615]

### System information

Godot v4.1.stable - Windows 10.0.19045 - Intel(R) Core(TM) i7-6700 CPU @ 3.40GHz (8 Threads)

### Issue description

Having infinite recursion "somewhere in the vicinity of" an exported variable with a custom getter makes the stack trace constantly refresh (?), lagging the editor and making it impossible to focus on any other tab or script.

https://github.com/godotengine/godot/assets/37083802/b322d919-1c11-487b-827f-bfe7c75f8779

### Steps to reproduce

```gdscript
extends Node

@export var my_var := 0:
	get:
		return my_var

func _ready() -> void:
	_ready()
```
This code reproduces it, but it happens even if the variable isn't part of the recursive script itself. The MRP includes an extra scene (`other_scene.tscn`) that causes the error merely for having a reference to a node with such a variable.

### Minimal reproduction project

[Stack bug test.zip](https://github.com/godotengine/godot/files/11994401/Stack.bug.test.zip)

**Comment by dalexeev** [comment]
> ```gdscript
> func _ready() -> void:
> 	_ready()
> ```

This causes an infinite recursion by itself, the `@export` variables have nothing to do with it.

> but it happens even if the variable isn't part of the recursive script itself. The MRP includes an extra scene (`other_scene.tscn`) that causes the error merely for having a reference to a node with such a variable.

`other_scene.gd` has the same error:

```gdscript
extends Node
@onready var node: Node = $Node # comment this out and the bug is gone

func _ready() -> void:
    _ready()
```

If you comment out `_ready()` there will be no error. Commenting out the `@onready` variable does not fix the error.

Just having a reference to the node with the error is not enough. Until the node is added to the tree, `_ready()` is not executed.

Please check if you have created the MRP correctly. Right now I don't see any Godot bug here.

**Comment by L4Vo5** [comment]
The error isn't the infinite recursion itself, but the editor being unresponsive because of it. Check the video I sent.

`other_scene.tscn` has a reference to its child, which a similar exported variable to `node.tscn`. If it used `node.tscn` directly, the recursion would come from `node.gd`, and it wouldn't demonstrate the issue happening in `other_scene.gd`.

**Comment by L4Vo5** [comment]
Here's another video, comparing how well the editor responds to normal recursion errors vs. how it does when the variable exists.
https://github.com/godotengine/godot/assets/37083802/a3e5542f-01bd-4b2d-904a-4bc6a3a0f99f

**Comment by dalexeev** [comment]
Thank you. Also, you cannot switch to other Debugger tabs (Errors, Profiler, etc.).

**Comment by vnen** [comment]
I can't reproduce this with current master (30f2a6d61). The editor stays responsive.

I tested on macOS, could this be a Windows-only bug?

**Comment by L4Vo5** [comment]
It seems to have been fixed somewhere between 4.2 dev1 and dev2
Tested both on Linux and Windows
